web/src/connectors/tasks-table-connector/use-job-filter.ts

```diff
@@ -18,29 +17,30 @@ export const useJobFilter = ({ fieldName }: Props) => {
 
     type PagingCache = {
         page: number;
-        cache: Array<Job>;
+        cache: Array<string>;
         search: string;
     };
 
     const loadJobs = createPagingCachedLoader(
         namesCache,
-        async (pageNumber, pageSize, keyword) => await jobsFetcher(pageNumber, pageSize, keyword)
+        async (pageNumber, pageSize, keyword) =>
+            await jobPropFetcher('name', pageNumber, pageSize, keyword)
     );
 
-    const jobNames = useLazyDataSource<Job, string, unknown>(
+    const jobNames = useLazyDataSource<string, string, unknown>(
         {
             api: loadJobs,
-            getId: (job) => job.id.toString()
+            getId: (job) => job.toString()
         },
         []
     );
 
-    const renderJobFilter = useColumnPickerFilter<Job, string, unknown, string>(
+    const renderJobFilter = useColumnPickerFilter<string, string, unknown, string>(
         jobNames,
         fieldName,
         {
             showSearch: true,
-            getName: (item) => (typeof item === 'boolean' ? String(item) : item.name)
+            getName: (item) => (typeof item === 'boolean' ? String(item) : item)
```

Comment:
I believe we don't need getName here, as we have only names. Please try without it

Reply:
Thanks for pointing it out. Removed